### PR TITLE
12 when failed an error gets displayed and the user is redirected to the register screen an error is that a username is already taken

### PR DESCRIPTION
**Pull request by serainaschraff**
if accepted I can do the same for login

### CODE REVIEW

File: src/components/views/Registration.js [hunk 1]
```diff
@@ -37,29 +38,38 @@ const Registration = () => {
...
 
       // Registration successfully worked --> navigate to the route /game in the GameRouter
       history.push(`/game`);
     } catch (error) {
-      alert(`Something went wrong during the registration: \n${handleError(error)}`);
+      const response = error.response;
+      if (response && `${response.status}`.toString() === "409") {
+        errorResponse = "Username already taken. Try again.\n";
+        console.log(errorResponse);
+        localStorage.setItem('errorMessage', errorResponse);
```
Comment (daveking7) [inline src/components/views/Registration.js:64]: Why do we have to save the error message?

**Review by daveking7 — APPROVED** [review]
Looks nice, if you know how to fix the movement of the register button (almost moves out of block), even better. But must wait for the implementation of the other fields from Joe.

File: src/components/views/Registration.js [hunk 2]
```diff
@@ -37,29 +38,38 @@ const Registration = () => {
...
+        window.location.assign(window.location);
```
[the remark this answers is not in the document]
Comment (serainaschraff) [reply src/components/views/Registration.js:65]: it reloads the page..

File: src/components/views/Registration.js [hunk 3]
```diff
@@ -37,29 +38,38 @@ const Registration = () => {
   const [username, setUsername] = useState("");
   const [name, setName] = useState("");
   const [password, setPassword] = useState("");
+  let errorResponse = null;
 
   const doRegister = async () => {
     try {
+      localStorage.removeItem('errorMessage');
```
[the remark this answers is not in the document]
Comment (serainaschraff) [reply src/components/views/Registration.js:45]: I don't see any other way of performing the task of providing the error message once the user tried to register with a taken username.. it is needed as later on the page is reloaded..

File: src/components/views/Registration.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (serainaschraff) [reply src/components/views/Registration.js:65]: see above

**Comment by serainaschraff** [comment]
> Looks nice, if you know how to fix the movement of the register button (almost moves out of block), even better. But must wait for the implementation of the other fields from Joe.

Yes I thought I will look at that issue again once Joe finished with his part.. (I'll make a new task so that we'll remember)

File: src/components/views/Registration.js (hunk 1 above)
Comment (serainaschraff) [reply src/components/views/Registration.js:64]: see above

**Pull request merged**